src/ert/analysis/_es_update.py

```diff
@@ -537,15 +511,25 @@ def analysis_ES(
                 f"No active observations for update step: {update_step.name}."
             )
 
-        smoother = ies.ES()
+        inversion_types = {0: "exact", 1: "subspace", 2: "subspace", 3: "subspace"}
+        inversion = inversion_types.get(module.ies_inversion, "Invalid input")
+        smoother_es = ies.ESMDA(
+            covariance=observation_errors**2,
+            observations=observation_values,
+            alpha=np.array([1]),
+            seed=rng,
+            inversion=inversion,
+        )
         truncation = module.enkf_truncation
         noise = rng.standard_normal(size=(num_obs, ensemble_size))
 
         if module.localization:
-            Y_prime = S - S.mean(axis=1, keepdims=True)
-            Sigma_Y = np.std(S, axis=1, ddof=1)
             batch_size: int = 1000
-            correlation_threshold = module.correlation_threshold(ensemble_size)
+            smoother_adaptive_es = AdaptiveESMDA(
```

Comment:
I'm sure you have this planned, but just so we don't forget, we should pass `cov_YY` to `AdaptiveESMDA` so it's not re-computed every time we call `assimilate`.

Reply:
I did not have this planned 👍